### ISSUE TITLE
Build error while executing tests/kernel/smp on ESP32

### ISSUE THREAD

**Issue by spoorthik** [zephyrproject-rtos/zephyr#12654]
**Describe the bug**
A build error is observed stating 'set_kernel_idle_time_in_ticks' defined but not used. 

**To Reproduce**
1. Setup  build environment for ESP32
2. cd zephyr.git/tests/kernel/smp
3. mkdir build; cd build
4. cmake -DBOARD=esp32
5. make flash
6. See error

**Screenshots or console output**
```
zephyr.git/kernel/idle.c:54:13: error: 'set_kernel_idle_time_in_ticks' defined but not used [-Werror=unused-function]

 static void set_kernel_idle_time_in_ticks(s32_t ticks)
             ^
cc1: all warnings being treated as errors
[ 65%] Building C object zephyr/kernel/CMakeFiles/kernel.dir/mempool.c.obj
zephyr/kernel/CMakeFiles/kernel.dir/build.make:88: recipe for target 'zephyr/kernel/CMakeFiles/kernel.dir/idle.c.obj' fa
make[3]: *** [zephyr/kernel/CMakeFiles/kernel.dir/idle.c.obj] Error 1
```

**Environment (please complete the following information):**
 - OS: Linux
 - Toolchain: Zephyr SDK, ESP32 toolchain
 - Commit SHA or Version used: 99aadec8576b9156c50d63e323733bccc15d5672

**Comment by andyross** [comment]
Yeah, I dropped a patch somewhere.  @dcpleung reported that esp32 doesn't build earlier (and I blew him off while dealing with timer stuff -- should have listened more carefully, sorry!), and in fact qemu_x86_64 remains blacklisted on that test when I know it was working.

Will fix.
